values/dev-stack-values.yaml

```diff
@@ -1 +1,4 @@
 extraPorts: ###ZARF_VAR_NGINX_EXTRA_PORTS###
+
+coreDnsOverrides: |
+  ###ZARF_VAR_COREDNS_OVERRIDES###
```

Comment:
Was the addition of the zarf var/values here primarily targetted at https://github.com/defenseunicorns/uds-k3d/issues/98? I think my preference might be to separate the two changes, i.e. just update the coredns configmap to use separate rewrite per gateway in this PR. 

I think the "different domain" warrants more discussion but my hope was that we could use the `DOMAIN` var that is common in most environments and possibly combine some actions to flow into the coredns config with the right value (i.e. we need to escape the `.`). I think that would be better than requiring the end user to recreate the full config block if we can avoid it. If there's other reasons to allow overriding this though I'd be interested to better understand the use cases.

Reply:
The CoreDNS overrides is more for other things that _may_ be installed into the cluster that require routing (e.g., passthrough gateway, if enabled?). The "different" domains part requires some more work, like CA/TLS cert management, etc. which I don't think we want to do in this dev environment haha.

Reply:
Gotcha yeah I wasn't even thinking about passthrough. That does throw a bit of a wrench in this since passthrough doesn't necessarily have a different domain (i.e. would be the same regex as `*.uds.dev`). But this helm override would cover it. I think this is okay to add in that case but I wonder if it would be better to split into two values? i.e. the configmap could look like:
```
  uds.override: |
    {{ .Values.coreDnsOverrides | nindent 4 }}
    {{ .Values.coreDnsDefaults | nindent 4 }}
```

Where `coreDnsDefaults` defaults to what you have in the values for tenant/admin and `coreDnsOverrides` does not have a default? It might simplify the end user experience of "adding another entry" (and I think we would want it first to handle cases like passthrough so it gets matched first?). Also would allow us to drop the defaults in the task/zarf.yaml since the zarf var would be for the "overrides" rather than "defaults".

Reply:
I have no issues with that, I can implement that